Assets/Scripts/Tags/Hazard.cs

```diff
@@ -1,45 +1,41 @@
 using UnityEngine;
+using UnityEngine.Serialization;
 
 public class Hazard : MonoBehaviour {
+    [Tooltip("Defines how high and low this object will bob up and down, measured in world-space units.")]
+    [SerializeField] private float amplitude = 0.2f;
 
-    [SerializeField] private float verticalSpeed;
-    [SerializeField] private float amplitude;
+    [Tooltip("The rate at which this object will rotate over time, measured in degrees per second.")]
+    [FormerlySerializedAs("rotSpeed")]
+    [SerializeField] private float angularSpeed = 60;
 
-    [SerializeField] private float rotSpeed;
+    [Tooltip("Defines the range within which a vertical speed will be randomly chosen for this object.\n\n" +
+        "This object's vertical speed describes the frequency of the up and down bobbing motion.")]
+    [SerializeField] private Vector2 verticalSpeedRange = new Vector2(2, 3);
 
-    private Vector3 origPos;
-    private Vector3 tempPos;
-
-    private Vector3 rot = new Vector3(45f, 45f, 45f);
-
-    private float rotDegreesPerSecond = 45f;
-
-    private static System.Random rndFloat = new System.Random();
-    private static System.Random rndRot = new System.Random();
-
-    private void Start () {
-        tempPos = origPos = new Vector3(transform.position.x, (float) rndFloat.NextDouble() + .3f, transform.position.z);
-        verticalSpeed = (float)rndFloat.NextDouble();
-
-        //rot = Quaternion.Euler((float)rndRot.NextDouble(), (float)rndRot.NextDouble(), (float)rndRot.NextDouble());
+    private Vector3 originalPosition;
+    private float verticalSpeed;
+    private Vector3 rotationalAxis;
 
+    private void Start() {
+        originalPosition = transform.position;
+        verticalSpeed = Random.Range(verticalSpeedRange.x, verticalSpeedRange.y);
+        rotationalAxis = Random.onUnitSphere;
     }
 
-    private void FixedUpdate() {
-        RandomFloating();
-        Rotation();
+    private void Update() {
+        UpdatePosition();
+        UpdateRotation();
     }
 
-    private void RandomFloating() {
-        tempPos = origPos;
-        tempPos.y += Mathf.Sin(Time.realtimeSinceStartup * (verticalSpeed + 2)) * amplitude;
```

Comment:
I used `Time.time` instead so if you had time scaling effects (slow-down time, pause, speed up time), these Hazard objects would move faster or slower as well.

`Time.realtimeSinceStartup` is useful when you NEED to measure real-life time, and DISREGARD time scaling effects.

Reply:
Hah! Always thinking ahead. I like that. Duly noted.

Okay. Yes, there's no need for us to measure how much time has passed since Start! That's true.